File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/itemService.js
//服务层
app.service('itemService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../item/findAll.php');
	};
	//分页 
	this.findByPagination=function(page,rows){
		return $http.get('../item/findByPagination.php?page='+page+'&rows='+rows);
	};
	//查询实体
	this.findById=function(id){
		return $http.get('../item/findById.php?id='+id);
	};
	//增加 
	this.add=function(entity){
		return  $http.post('../item/add.php',entity );
	};
	//修改 
	this.update=function(entity){
		return  $http.post('../item/update.php',entity );
	};
	//删除
	this.dele=function(ids){
		return $http.get('../item/delete.php?ids='+ids);
	};
	//搜索
	this.findWithConditionsByPagination=function(page,rows,searchEntity){
		return $http.post('../item/findWithConditionsByPagination.php?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/cn/porkchop/pinyougou/sellergoods/service/ItemService.java
package cn.porkchop.pinyougou.sellergoods.service;

import cn.porkchop.pinyougou.pojo.PageResult;
import cn.porkchop.pinyougou.pojo.TbItem;

import java.util.List;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ItemService {

	/**
	 * 返回全部列表
	 * @return
	 */
    List<TbItem> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
    PageResult findByPagination(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
    void add(TbItem item);
	
	
	/**
	 * 修改
	 */
    void update(TbItem item);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
    TbItem findById(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
    void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
    PageResult findWithConditionsByPagination(TbItem item, int pageNum, int pageSize);
	
}
